injector/src/main/java/edu/ucr/cs/riple/injector/location/LocationToJsonVisitor.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2023 Nima Karimipour
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.injector.location;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+/**
+ * A visitor that converts a location to a JSON object. This visitor is singleton and for all
+ * methods {@link LocationToJsonVisitor#INSTANCE} should be used.
+ */
+public class LocationToJsonVisitor implements LocationVisitor<JSONObject, Void> {
+
+  /** The Keys used to represent a location in JSON format */
+  public enum KEYS {
+    VARIABLES,
+    METHOD,
+    KIND,
+    CLASS,
+    PATH,
+    INDEX
+  }
+
+  /** A singleton instance of this visitor. */
+  public static final LocationToJsonVisitor INSTANCE = new LocationToJsonVisitor();
```

Comment:
Why make this a singleton?  Since there is no state in the object, and allocating the object should be cheap, I think this is overkill / premature optimization.

Reply:
Thank you for noticing this. I thought since there is not state, we should make it singleton. But as you said, this is cheap and not a good optimization overall. I fixed it here af081c111fe9b8c6dbb2fc22ea657027a93c35cf.